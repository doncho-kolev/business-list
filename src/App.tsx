import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route,
    Outlet,
} from "react-router-dom";
import Business from './components/business/Business';
import NotFound from './components/notFound/NotFound';
import Layout from './components/Layout';
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';
import { BusinessLink } from './services/useGetBusinesses';
import AllBusiness from './components/business/AllBusinesses';

const client = new ApolloClient({
    cache: new InMemoryCache(),
    link: BusinessLink
});

function App() {
    return (
        <Router >
            <Routes>
                <Route element={<Layout><Outlet /></Layout>}>
                    <Route index element={<AllBusiness />} />
                    <Route path="/business/:id" element={<Business />} />
                    <Route path="*" element={<NotFound />} />
                </Route>
            </Routes>
        </ Router>
    );
}

const ApolloApp = () => (
    <ApolloProvider client={client}>
        <App />
    </ApolloProvider>
);

export default ApolloApp;