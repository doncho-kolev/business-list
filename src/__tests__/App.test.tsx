import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../App';

test('renders learn react link', () => {
    const mount = render(<App />);
    expect(mount.asFragment().children[0]).toMatchSnapshot();
});
