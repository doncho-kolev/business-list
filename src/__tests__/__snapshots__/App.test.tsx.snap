// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="app-container"
>
  <div
    class="logo"
  >
    Logo
  </div>
  <div>
    <p>
      Loading...
    </p>
  </div>
</div>
`;
